src/components/DocumentManager/hooks/DocumentManagerState.js

```diff
@@ -6,35 +6,41 @@ import {
 
 function useDocumentManagerState(collectionId) {
   let [documents, setDocuments] = useState(null);
+  // TODO: Suggest storing only the selectedDocument's id here. Doing so avoids having the full data in two spots, which reduces memory usage and avoids the risk of out of sync
+  // between the two datasets.
   let [selectedDocument, setSelectedDocument] = useState(null);
 
   useEffect(() => {
     (async () => {
-      const collectionDocuments = await getCollectionDocuments(collectionId);
-      setDocuments(collectionDocuments);
-
-      if (collectionDocuments.length > 0) {
-        const documentWithFile = await getDocument(collectionDocuments[0].id);
-        updateCollectionDocument(collectionDocuments, documentWithFile);
-        setSelectedDocument(documentWithFile);
+      try {
+        const docs = await getCollectionDocuments(collectionId);
+        if (docs.length > 0) {
+          const doc = await getDocument(docs[0].id);
+          updateState(docs, doc);
+        }
+      } catch (error) {
+        alert("Loading documents failed. Please try reloading your browser.");
```

Comment:
Note that I'm calling try/catch here. Important to do so with async/await for error handling (or can use .catch since it uses promises under the hood).

If you set Chrome to offline after initial load and click a different document, you'll see that the alert displays. Suggest using a toast instead of my hacked in alert. Perhaps [this](https://reactstrap.github.io/components/alerts/) or [this](https://fkhadra.github.io/react-toastify/).

Reply:
Did not implement try/catches for same reason as above